head/src/test/kotlin/io/qalipsis/core/head/inmemory/StandaloneCampaignManagerTest.kt

```diff
@@ -416,6 +420,103 @@ internal class StandaloneCampaignManagerTest {
         }
     }
 
+    @Test
+    internal fun `should replay a campaign `() = testDispatcherProvider.run {
+        // given
+        val campaignManager = standaloneCampaignManager(this)
```

Comment:
You can use a spied standaloneCampaignManager, and just record the call to start and the retrieval of the campign from the CampaignService. It will make the test much easier and more robust. (Same for the Redis-based implementation)